### PR TITLE
Update middleware.rb

### PR DESCRIPTION
**Pull request by washu**
convert all constant strings, to froze const vars.
performance improvement for the gc cycles. it also uses ruby 2.4 match? for faster regex tests if available

### CODE REVIEW

File: lib/message_bus/rack/middleware.rb [hunk 1]
```diff
@@ -56,25 +56,66 @@ def self.backlog_to_json(backlog)
...
+  NO_SNIFF = 'nosniff'.freeze
+  TRANSFER_ENCODING = 'Transfer-Encoding'.freeze
+  CHUNKED = 'chunked'.freeze
+  TEXT_CONTENT_TYPE_LINE = 'text/plain; charset=utf-8'.freeze
+  EMPTY_LIST = "[]".freeze
+  
   def call(env)
 
-    return @app.call(env) unless env['PATH_INFO'] =~ /^\/message-bus\//
+    return @app.call(env) unless USE_MATCH ? BUS_REGEX.match?(env[PATH_INFO]) : env[PATH_INFO] =~ BUS_REGEX
```
Comment (SamSaffron) [inline lib/message_bus/rack/middleware.rb]: hmmm no regex is even needed here, can simply use `start_with?`

File: lib/message_bus/rack/middleware.rb [hunk 2]
```diff
@@ -107,27 +148,27 @@ def call(env)
...
     end
 
-    if env["REQUEST_METHOD"] == "OPTIONS"
-      return [200, headers, ["OK"]]
+    if env[REQUEST_METHOD] == OPTIONS
+      return [200, headers, [OK]]
     end
 
     long_polling = @bus.long_polling_enabled? &&
-                   env['QUERY_STRING'] !~ /dlp=t/.freeze &&
```
Comment (SamSaffron) [inline lib/message_bus/rack/middleware.rb]: no need for regex here as well.

**Comment by SamSaffron** [comment]
Overall I feel this makes the code a bit less clear, instead perhaps add 

```
# frozen_string_literal: true
```

Then we only need constants for stuff that is used more than once, the indirection introduces some complexity.

**Comment by washu** [comment]
frozen_string_literal is a ruby 2.3+ feature only and is a pragma directive

**Comment by SamSaffron** [comment]
agree, but I think it is fair to provide this perf boost only to 2.3+

**Comment by washu** [comment]
in that case i can abandon the pull request, if you want to just go ahad and add the 3 lines that would be left. the 2 starts with and the comment

**Pull request closed without merging**